### PR TITLE
Add BaseApp events support

### PR DESCRIPTION
**Pull request by grozdanov**
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

Added support for `onAppStart`, `onAppPause` and `onAppResume` BaseApp events by using `Scene` component.
```
<Scene
  onAppStart={(data) => console.log(data)}
  onAppPause={() => console.log('onAppPause')}
  onAppResume={() => console.log('onAppResume')}
>
...
</Scene>
```

### CODE REVIEW

File: src/platform/lumin-runtime/mxs-base-app.js [hunk 1]
```diff
@@ -4,16 +4,37 @@ import { AppPrismController } from './controllers/app-prism-controller.js';
...
 
-  onAppStart(arg) {
+  get OnAppStartData () {
+    return this._onAppStartData;
+  }
+
+  onAppStart(args) {
+    this._onAppStartData = { 
+      uri: args.getUri(),
+      isInternetConnected: typeof this.isInternetConnected === 'function' ? this.isInternetConnected() : undefined,
```
Comment (tonycuadra) [inline src/platform/lumin-runtime/mxs-base-app.js]: Everything besides uri seems unrelated to app start data. Also, besides isShareableApp (which I believe reflects a manifest field), the rest could change at any moment so it seems problematic to take a snapshot of them only when onAppStart is called.

Comment (grozdanov) [reply src/platform/lumin-runtime/mxs-base-app.js]: I agree! The connectivity might change. There is an event `WifiStatusEvent` which the developer can use to get updates on the connectivity state. The goal was to provide some initial app status which there is no other way to provide until an event occurs.

Comment (tonycuadra) [reply src/platform/lumin-runtime/mxs-base-app.js]: This seems unintuitive but I don't know the best way to handle this in React so I'll resolve the change request.

Comment (pleszkiewicz) [reply src/platform/lumin-runtime/mxs-base-app.js]: In iOS, there is `Reachability` framework that notifies about WiFi/Bluetooth/Cell network changes. We could provide a similar module as we did with Platform. But this is another task, isn't it?
